src/Microsoft.DotNet.Configurer/NuGetCachePrimer.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.Configurer
+{
+    public class NuGetCachePrimer : INuGetCachePrimer
+    {
+        private const string NUGET_SOURCE_PARAMETER = "-s";
+        private readonly ICommandFactory _commandFactory;
+        private readonly IDirectory _directory;
+        private readonly IFile _file;
+        private readonly INuGetPackagesArchiver _nugetPackagesArchiver;
+        private readonly INuGetCacheSentinel _nuGetCacheSentinel;
+
+        public NuGetCachePrimer(
+            ICommandFactory commandFactory,
+            INuGetPackagesArchiver nugetPackagesArchiver,
+            INuGetCacheSentinel nuGetCacheSentinel)
+            : this(commandFactory,
+                nugetPackagesArchiver,
+                nuGetCacheSentinel,
+                FileSystemWrapper.Default.Directory,
+                FileSystemWrapper.Default.File)
+        {
+        }
+
+        internal NuGetCachePrimer(
+            ICommandFactory commandFactory,
+            INuGetPackagesArchiver nugetPackagesArchiver,
+            INuGetCacheSentinel nuGetCacheSentinel,
+            IDirectory directory,
+            IFile file)
+        {
+            _commandFactory = commandFactory;
+            _directory = directory;
+            _nugetPackagesArchiver = nugetPackagesArchiver;
+            _nuGetCacheSentinel = nuGetCacheSentinel;
+            _file = file;
+        }
+
+        public void PrimeCache()
+        {
+            if(SkipPrimingTheCache())
+            {
+                return;
+            }
+
+            var pathToPackagesArchive = _nugetPackagesArchiver.ExtractArchive();
+
+            PrimeCacheUsingArchive(pathToPackagesArchive);
+        }
+
+        private bool SkipPrimingTheCache()
+        {
+            return !_file.Exists(_nugetPackagesArchiver.NuGetPackagesArchive);
```

Comment:
This is for stage1. I didn't want to generate the archive in stage1 as well because it takes a bit for this to happen. So I put this here to skip priming when the archive is not found. I created a test stating this scenario. I didn't want to call this SkipPrimingTheCacheInStage1. What do you think? Remove this and increase our build time even more? Change the name here?